pkg/apis/org/conversion.go

```diff
@@ -0,0 +1,193 @@
+		err := yaml.Unmarshal([]byte(v1Data), &v1Spec)
+		if err != nil {
+			return err
+		}
+	}
+
+	v1Obj.ObjectMeta = v2.ObjectMeta
+	v1Obj.Spec = v1Spec
+	v1Obj.Status = v1.CheClusterStatus{}
+
+	v2Spec, err := yaml.Marshal(v2.Spec)
+	if err != nil {
+		return err
+	}
+	if v1Obj.Annotations == nil {
+		v1Obj.Annotations = map[string]string{}
+	}
+	v1Obj.Annotations[v2alpha1StorageAnnotation] = string(v2Spec)
+
+	v2alpha1ToV1_Host(v1Obj, v2)
+	v2alpha1ToV1_WorkspaceBaseDomain(v1Obj, v2)
+	v2alpha1ToV1_GatewayEnabled(v1Obj, v2)
+	v2alpha1ToV1_GatewayImage(v1Obj, v2)
+	v2alpha1ToV1_GatewayConfigurerImage(v1Obj, v2)
+	v2alpha1ToV1_TlsSecretName(v1Obj, v2)
+	v2alpha1ToV1_K8sIngressAnnotations(v1Obj, v2)
+
+	return nil
+}
+
+func v1ToV2alpha1_Host(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	v2.Spec.Gateway.Host = v1.Spec.Server.CheHost
+}
+
+func v1ToV2alpha1_WorkspaceBaseDomain(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	if util.IsOpenShift {
+		v2.Spec.WorkspaceBaseDomain = v1.Spec.Server.CustomCheProperties[routeDomainSuffixPropertyKey]
+	} else {
+		v2.Spec.WorkspaceBaseDomain = v1.Spec.K8s.IngressDomain
+	}
+}
+
+func v1ToV2alpha1_GatewayEnabled(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	val := util.GetServerExposureStrategy(v1) == "single-host"
+	v2.Spec.Gateway.Enabled = &val
+}
+
+func v1ToV2alpha1_GatewayImage(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	v2.Spec.Gateway.Image = v1.Spec.Server.SingleHostGatewayImage
+}
+
+func v1ToV2alpha1_GatewayConfigurerImage(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	v2.Spec.Gateway.ConfigurerImage = v1.Spec.Server.SingleHostGatewayConfigSidecarImage
+}
+
+func v1ToV2alpha1_TlsSecretName(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	// v1.Spec.Server.CheHostTLSSecret is used specifically for Che Host - i.e. the host under which che server is deployed.
+	// In DW we would only used that for subpath endpoints but wouldn't know what TLS to use for subdomain endpoints.
+	// Che server always uses the default cluster certificate for these on OpenShift, and the K8s.TlsSecretName on K8s.
+	// Because we're dealing with endpoints, let's try to use the secret on Kubernetes and nothing (e.g. the default cluster cert on OpenShift)
+	// which is in line with the logic of the Che server. Let's ignore CheHostTLSSecret for now - this is used only for che server which
+	// we are not exposing anyway.
+	if !util.IsOpenShift {
+		v2.Spec.TlsSecretName = v1.Spec.K8s.TlsSecretName
+	}
+}
+
+func v1ToV2alpha1_K8sIngressAnnotations(v1 *v1.CheCluster, v2 *v2alpha1.CheCluster) {
+	annotations := map[string]string{
+		"kubernetes.io/ingress.class":                       v1.Spec.K8s.IngressClass,
+		"nginx.ingress.kubernetes.io/proxy-read-timeout":    "3600",
+		"nginx.ingress.kubernetes.io/proxy-connect-timeout": "3600",
+		"nginx.ingress.kubernetes.io/ssl-redirect":          "true",
+	}
+	// This is what is applied in the deploy/ingress.go but I don't think it is applicable in our situation
```

Comment:
To be honest I don't know either. 
If registries will be accessible without that annotation then it is fine for me.